lib/puppet/provider/service/launchd.rb

```diff
@@ -181,10 +194,9 @@ def self.job_list
   # it is 10.6 or greater. This allows us to implement different plist
   # behavior for versions >= 10.6
   def has_macosx_plist_overrides?
-    @product_version ||= self.class.get_macosx_version_major
     # (#11593) Remove support for OS X 10.4 & earlier
     # leaving this as is because 10.5 still didn't have plist support
-    return true unless /^10\.[0-5]/.match(@product_version)
```

Comment:
I think this function would be just as clear if it said `return self.class.get_os_version >= 9`. The comments are no longer as clear however, os get_os_version is returning Darwin versions.

Also, I'm not sure the comparison is correct, as according to https://en.wikipedia.org/wiki/Darwin_(operating_system) OS X 10.5 was Darwin 9.